backends/sqlite/sqlite_backend.go

```diff
@@ -0,0 +1,768 @@
+	}
+
+	ctx = context.WithValue(ctx, txCtxVarKey, tx)
+
+	err = s.updateJob(ctx, jobErr, internal.JobStatusInProgress)
+	if err != nil {
+		if errors.Is(err, context.Canceled) {
+			return
+		}
+		err = fmt.Errorf("error updating job status: %w", err)
+		return err
+	}
+	err = tx.Commit()
+	if err != nil {
+		errMsg := "unable to commit job transaction. retrying this job may duplicate work:"
+		s.logger.Error(errMsg, slog.String("queue", h.Queue), slog.Any("error", err), slog.Int64("job_id", job.ID))
+		_ = tx.Rollback()
+		return fmt.Errorf("%s %w", errMsg, err)
+	}
+	return
+}
+
+func (s *SqliteBackend) getJob(ctx context.Context, jobID string) (job *jobs.Job, err error) {
+	job = &jobs.Job{}
+	var payload string
+
+	row := s.db.QueryRowContext(ctx, JobQuery, jobID)
+	err = row.Scan(
+		&job.ID, &job.Fingerprint, &job.Queue, &job.Status, &job.Deadline,
+		&payload, &job.Retries, &job.MaxRetries,
+		&job.RunAfter, &job.RanAt, &job.CreatedAt, &job.Error,
+	)
+	if err != nil {
+		return
+	}
+
+	if err = json.Unmarshal([]byte(payload), &job.Payload); err != nil {
+		s.logger.Error("err getting job", slog.String("err", err.Error()))
+		return
+	}
+
+	return
+}
+
+// withJobContext creates a new context with the Job set
+func withJobContext(ctx context.Context, j *jobs.Job) context.Context {
+	return context.WithValue(ctx, internal.JobCtxVarKey, j)
+}
+
+// updateJob updates the status of jobs with: status, run time, error messages, and retries
+//
+// if the retry count exceeds the maximum number of retries for the job, move the job to the dead jobs queue
+//
+// if `tx`'s underlying connection dies while updating job status, the transaction will fail, and the job's original
+// status will be reflecting in the database.
+//
+// The implication of this is that:
+// - the job's 'error' field will not reflect any errors the occurred in the handler
+// - the job's retry count is not incremented
+// - the job's run time will remain its original value
+// - the job has its original 'status'
+//
+// ultimately, this means that any time a database connection is lost while updating job status, then the job will be
+// processed at least one more time.
+func (s *SqliteBackend) updateJob(ctx context.Context, jobErr error, status string) (err error) {
+	errMsg := ""
+
+	if jobErr != nil {
+		s.logger.Error("job failed", slog.Any("job_error", jobErr))
+		status = internal.JobStatusFailed
+		errMsg = jobErr.Error()
+	}
+
+	var job *jobs.Job
+	if job, err = jobs.FromContext(ctx); err != nil {
+		return fmt.Errorf("error getting job from context: %w", err)
+	}
+
+	var tx *sql.Tx
+	if tx, err = txFromContext(ctx); err != nil {
+		return
+	}
+
+	if job.MaxRetries != nil && job.Retries >= *job.MaxRetries {
+		err = s.moveToDeadQueue(ctx, job, errMsg)
+		return
+	}
+
+	// In progress job
+	if jobErr == nil && len(status) != 0 {
+		qstr := "UPDATE neoq_jobs SET status = $1, error = $2 WHERE id = $3"
+		_, err = tx.ExecContext(ctx, qstr, status, errMsg, job.ID)
```

Comment:
Let's check for this error and return an error when it's non-nil. Otherwise we could have in-progress jobs without an in-progress `status`, which may cause confusion.

Reply:
It is captured in line 440 to 447 right? That `err` variable gets returned from the function.

Reply:
I see, my mistake.